### PR TITLE
feat(persistence): add PostgreSQL database persistence layer

### PR DESCRIPTION
**Pull request by dukex**
## Overview

This PR implements a comprehensive PostgreSQL persistence layer for Operion, providing production-ready database storage for workflows, triggers, and steps.

### Core Implementation
- **Normalized Database Schema** - Separate tables for workflows, workflow_triggers, and workflow_steps
- **Automated Migrations** - Database schema automatically created and updated on startup
- **Schema Versioning** - Uses schema_migrations table to track applied migrations  
- **Soft Delete Support** - Workflows soft deleted using deleted_at timestamp
- **JSONB Configuration Storage** - Complex configuration data stored as JSONB
- **Transaction Safety** - All migrations and operations run within database transactions
- **Modular Architecture** - Clean separation with migration.go, workflow.go, and postgres.go

### Database Schema
- **workflows** table stores core workflow data (id, name, description, variables, metadata, status, timestamps)
- **workflow_triggers** table stores trigger definitions with foreign key to workflows
- **workflow_steps** table stores step definitions with foreign key to workflows and unique constraint on workflow_id + uid
- **schema_migrations** table tracks migration versions and timestamps
- **UUID v7 Support** - All table IDs use time-ordered UUID v7 with auto-generation for better performance and natural sorting
- Comprehensive indexes on foreign keys, status, owner, creation time, and deletion timestamp
- Cascade deletes ensure referential integrity

### Repository Pattern
- **WorkflowRepository** handles all workflow CRUD operations
- Supports complex operations with triggers and steps in single transactions
- Automatic loading of related triggers and steps when retrieving workflows
- Efficient bulk operations for saving/updating workflow components

## Configuration

The PostgreSQL persistence layer can be configured using the DATABASE_URL environment variable:

```bash
# PostgreSQL connection
DATABASE_URL=postgres://user:password@localhost:5432/operion?sslmode=disable

# File-based storage (existing)
DATABASE_URL=./data/workflows
```

### New Files
- `pkg/persistence/postgresql/postgres.go` - Main persistence implementation
- `pkg/persistence/postgresql/workflow.go` - Workflow repository with CRUD operations  
- `pkg/persistence/postgresql/migration.go` - Database migration management
- `pkg/persistence/postgresql/postgres_test.go` - Comprehensive test suite
- `pkg/persistence/sqlbase/migration.go` - Shared migration framework

### Modified Files
- `pkg/cmd/persistence.go` - Added PostgreSQL persistence factory
- `README.md` - Updated with PostgreSQL configuration documentation
- `CLAUDE.md` - Added PostgreSQL implementation details

## Testing

- **510 lines of comprehensive tests** covering all CRUD operations
- Integration tests with transaction rollback
- Migration testing with version tracking
- Error handling and edge case coverage
- Connection health checks

## Migration Path

This implementation maintains full backward compatibility with existing file-based storage. Users can:

1. Continue using file-based storage (no changes required)
2. Migrate to PostgreSQL by updating DATABASE_URL
3. Switch between persistence backends without code changes

## Production Ready Features

- Connection pooling and health checks
- Proper error handling with wrapped errors
- Structured logging with slog
- Transaction-safe operations
- Efficient indexing strategy
- Automatic schema versioning
- Soft delete functionality

### CODE REVIEW

File: pkg/persistence/postgresql/postgres.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	"log/slog"
+
+	"github.com/dukex/operion/pkg/models"
+	"github.com/dukex/operion/pkg/persistence/sqlbase"
+
+	_ "github.com/lib/pq"
+)
+
+// Persistence implements the persistence layer for PostgreSQL.
+type Persistence struct {
```
[the remark this answers is not in the document]
Comment (dukex) [reply pkg/persistence/postgresql/postgres.go:17]: The Effective Go has some advise about Naming https://go.dev/doc/effective_go#names

> exported: type name will be used as postgres.PostgresPersistence by other packages, and that stutters; consider calling this Persistence (revive)

The ideia have something like that `postgres.Persistence` instead of `postgres.PostgresPersistence`
